src/main/scala/rps/RPSMoves.scala

```diff
@@ -0,0 +1,37 @@
+package rps
+
+
+/*
+
+Scala has a strange view of Enumerations, that is
+a little bit too much for the average use-case.
+
+The "agile-version" of enumerations are called
+"sealed trait extended by case objects".
+
+Here we define a RPSMoves ADT that will contain
+useful definitions and methods to craft RPS Moves.
+
+*/
+
+
+object RPSMoves {
+
+sealed abstract class RPSMove(
+    val icon : String,
+    val value : String) {
+
+    override def toString = icon
```

Comment:
About attaching "behavioral" data to the class, I suggest you to read my comment to Fede's solution: https://github.com/buildo/rockPaperScissorByFede/pull/9#discussion_r201639206

You can ignore it for now, but it's good to know ;)

Reply:
At a super high-level, you want to have pure data (no behavior attached) and have a separate function for producing a behavior. For example:

```scala
def printMove(move: Move): String = move match {
  case Rock => "rock"
  case Paper => "paper"
  // ...
}
```

As I mentioned in the other comment, this is the foundation of ad-hoc polymorphism, which leads to typeclasses. The answer to your specific question is: use a typeclass. A good candidate for printing is `Show` (which is defined by many FP libraries), but more often you need to transform the value into a JSON string or similar, so you use different typeclasses according to the specific library you're using (we'll see an example in the next steps)

Reply:
Gotcha! So the answer is typeclasses :3 Very good!